web/static/js/content_types/image_editor.js

```diff
@@ -56,37 +57,78 @@ class ImageEditor {
   }
 
   onSubmit (data) {
-    const editor = document.querySelector(`[data-thesis-content-id="${data.contentId}"`)
+    this.set(data.contentId, data)
+    this.closeTray()
+  }
+
+  getContent (ed) {
+    const type = ed.getAttribute('data-thesis-content-type')
+    if (type === 'image') {
+      return ed.querySelector('img').getAttribute('src')
+    } else {
+      return this.getUrlFromStyle(ed.style.backgroundImage)
+    }
+  }
+
+  uploadAndSet (data, page, callback) {
+    const imageUrl = this.determineImageUrl(data.content, page.origin)
+    if (!imageUrl) { callback(); return }
+
+    Net.post('/thesis/files/import', {image_url: imageUrl, image_name: data.name})
+    .then((response) => {
+      callback()
+      if (response.path.length > 0) this.set(data.name, {url: response.path}, data.meta)
+    })
+    .catch((error) => { callback(); window.alert(error) })
+  }
+
+  set (name, data, meta = null) {
+    const editor = document.querySelector(`[data-thesis-content-id="${name}"`)
     editor.classList.add('modified')
 
-    const meta = JSON.stringify({alt: data.alt})
+    if (!meta) {
+      const prevMeta = JSON.parse(editor.getAttribute('data-thesis-content-meta'))
+      const newMeta = Object.assign({}, prevMeta, {alt: data.alt})
+      meta = JSON.stringify(newMeta)
+    }
+
     editor.setAttribute('data-thesis-content-meta', meta)
 
     const type = editor.getAttribute('data-thesis-content-type')
     if (type === 'image') {
       const img = editor.querySelector('img')
       img.src = data.url
-      img.alt = data.alt
+      img.alt = JSON.parse(meta).alt
```

Comment:
since we're not getting alt text during import, we have to access it from the meta. this line works for both cases